airgun/views/ansible_role.py

```diff
@@ -0,0 +1,75 @@
+from widgetastic.widget import Checkbox
+from widgetastic.widget import Text
+from widgetastic_patternfly import BreadCrumb
+from widgetastic_patternfly4 import Button
+from widgetastic_patternfly4 import PatternflyTable
+
+from airgun.views.common import BaseLoggedInView
+from airgun.views.common import SatTable
+from airgun.views.common import SearchableViewMixin
+from airgun.widgets import ActionsDropdown
+from airgun.widgets import Pagination
+
+
+class ImportPagination(Pagination):
+    PER_PAGE_BUTTON_DROPDOWN = ".//div[button[@id='pagination-options-menu-toggle-2']]"
+
+
+class AnsibleRolesView(BaseLoggedInView, SearchableViewMixin):
+    """Main Ansible Roles view. Prior to importing any roles, only the import_button
+    is present, without the search widget or table.
+    """
+
+    title = Text("//h1[contains(., text()='Ansible Roles')")
+    import_button = Text("//a[contains(@href, '/ansible_roles/import')]")
+    submit = Button('Submit')
+    total_imported_roles = Text("//span[@class='pagination-pf-items-total']")
```

Comment:
From the xpath, this looks like the pagination widget's `total_items` attribute:
https://github.com/SatelliteQE/airgun/blob/master/airgun/widgets.py#L1453

You can reference it straight through `view.pagination.total_items` which I generally recommend, but if you want to create a more direct reference to the number of imported roles you could define a small property function.

```
@property
def total_imported_roles(self):
    return self.pagination.total_items
```

Reply:
Been trying to figure out how to implement this. I see a couple of tests in Robottelo that use do something similar, such as test_contenthost.py:

https://github.com/SatelliteQE/robottelo/blob/9edf53cb6f30c77e13ab4557b1668033406904c1/tests/foreman/ui/test_contenthost.py#L1321

So it looks like I'll need to add something like the `read_all` method from the content host entity in order to access that data:

https://github.com/SatelliteQE/airgun/blob/e6a29c3c0a46dc8a63c9818477fc733ae3a613fb/airgun/entities/contenthost.py#L31

Is something like this what you had in mind, or are there other possibilities?

Reply:
To access this value through the entity itself, yes you would need an entity function like `read_all` to read the view and return the dict.

The Pagination widget itself can just be directly read, or its `total_items` property can be accessed.